### ISSUE TITLE
Mermaid quadrantChart (and experimental charts) are not working

### ISSUE THREAD

**Issue by JaneX8** [laurent22/joplin#8728]
Both examples are from the Mermaidjs website (https://mermaid.js.org/syntax/quadrantChart.html) but don't work in the Joplin Android app.

```mermaid
quadrantChart
  x-axis Urgent --> Not Urgent
  y-axis Not Important --> "Important ❤"
  quadrant-1 Plan
  quadrant-2 Do
  quadrant-3 Delegate
  quadrant-4 Delete
```

and 

```mermaid
quadrantChart
    title Reach and engagement of campaigns
    x-axis Low Reach --> High Reach
    y-axis Low Engagement --> High Engagement
    quadrant-1 We should expand
    quadrant-2 Need to promote
    quadrant-3 Re-evaluate
    quadrant-4 May be improved
    Campaign A: [0.3, 0.6]
    Campaign B: [0.45, 0.23]
    Campaign C: [0.57, 0.69]
    Campaign D: [0.78, 0.34]
    Campaign E: [0.40, 0.34]
    Campaign F: [0.35, 0.78]
```

Result in "Syntax error in graph
mermaid version 9.2.2".

![Screenshot_2023-08-25-15-07-06-45_97a2b898327add53e982bbebadc3de52.jpg](https://github.com/laurent22/joplin/assets/5116641/fe445f35-8072-417a-ab24-50d373fe494c)

I also noticed that other experimental charts like Timeline and Mindmaps don't work in Joplin but I guess those are simple not available yet in Joplin. It would be great if we could enable experimental support for Mermaid in the app too.

## Environment

Joplin version: 2.11.32 (db v42)
Platform: Android
OS specifics: OxygenOS (OnePlus)

**Comment by oj-lappi** [comment]
The timeline chart is available since mermaid version 9.4, and Joplin is using 9.2.2, I assume the fix is to bumpi the version number in the package.json file:

https://github.com/laurent22/joplin/blob/75e6ae135b2f8b917426fe17e808419583acf5cf/packages/renderer/package.json#L52

I'm not sure if there are any rendering tests in Joplin, or if there are compatibility issues, but naively I would assume staying within the same major version should be safe at least.

---

UPDATE: Tested this out on a checked out repo and it seems fairly straightforward, bump the version number, run `yarn install` and `Tools/buildAssetr.js` to copy over a minified version from the mermaid package, and we get the new charts.

I'll run some tests and see if I can put this into a pull request.

**Comment by Loqova** [comment]
Great! Thank you. When will this be available in the Play Store Joplin app? I still have the same 9.2.2 issue as shown above in

```
Joplin: 2.12.3
Database: v43
Platform: Android
OS specifics: OxygenOS (OnePlus)
```
